CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.12.0)
 
-project(ni_grpc_device_server C CXX)
+project(ni_grpc_device_server VERSION 1.5.2.0 LANGUAGES C CXX)
```

Comment:
Updating this should be added to the [release process page](https://github.com/ni/grpc-device/wiki/Release-Process), at least. But preferably, we should figure out a way to automate it. (Can we pass in the version as a variable to cmake, but maybe default it to 0.0.0.0? And if so, do we have the version number available in our CI builds to be able to pass down to cmake?)

Reply:
`CMakeLists.txt` can set the version based on a variable, which you could pass in on the command line, read from a file, or read directly from Git. Here's an SO Q/A with some links: https://stackoverflow.com/questions/53551648/shared-library-versioning-with-cmake-on-github

Also, a GitHub action can update a file based on the git tag. This is from one of our other repos which isn't public yet:
```
      # If the tag is 0.1.0, this will set the version of the package to 0.1.0
      - name: Store version from Tag
        id: vars
        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}

      - name: Promote the package version to tag name.
        run: |
          poetry version ${{ steps.vars.outputs.tag }}
```

Reply:
Updated CMakeList to parse the current branch's git tag. Use the parsed `GIT_TAG_MAJOR`, `GIT_TAG_MINOR`, and `GIT_TAG_PATCH` variables to generate version resource file.

Reference examples:
https://github.com/pmirshad/cmake-with-git-metadata/blob/master/CMakeLists.txt
https://ipenguin.ws/2012/11/cmake-automatically-use-git-tags-as.html